apps/cowswap-frontend/src/modules/trade/hooks/setupTradeState/useSetupTradeStateFromUrl.ts

```diff
@@ -23,7 +23,7 @@ export function useSetupTradeStateFromUrl(): null {
   const tradeStateRef = useRef(state)
   tradeStateRef.current = state
 
-  useEffect(() => {
+  useMemo(() => {
```

Comment:
i don't get it. Why useMemo? you don't return anything and you are triggering some side effects in the memo

Reply:
Can you add that explanation to the code?
Otherwise I can see myself replacing that with a useEffect next time I touch this code.

Reply:
You caught me :)
It covers all the mess I did in `useSetupTradeState`.
This hook is full of missind deps in `useEffect` "by design".
But it will be better to use refs instead.
One day I will refactor it...

Reply:
👍🙏